src/Talis/Persona/Client/Tokens.php

```diff
@@ -113,21 +111,24 @@ public function retrieveJWTCertificate($cacheTTL = 300)
      * Validate the given token by using Persona
      * @param array $params a set of optional parameters you can pass to this method <pre>
      *      access_token: (string) a token to validate explicitly, if you do not specify one the method tries to find one,
-     *      scope: (string) specify this if you wish to validate a scoped token
+     *      scopes: (array) specify this if you wish to validate a scoped token
      * @return bool|string will return false if could not validate the token, else true
      * $throws \Exception if you do not supply a token AND it cannot extract one from $_SERVER, $_GET, $_POST
      */
-    protected function validateTokenUsingPersona($token, $scope)
+    protected function validateTokenUsingPersona($token, $scopes)
     {
         // verify against persona
         $this->getStatsD()->increment('validateToken.cache.miss');
         $url = $this->getPersonaHost() . $this->config['persona_oauth_route'] . '/' . $token;
 
-        if (empty($scope) === false) {
-            $url .= '?scope=';
-            $url .= $scope !== 'su'
-                ? 'su,' . $scope
-                : $scope;
+        if (empty($scopes) === false) {
+            $suKey = array_search('su', $scopes, true);
+
+            if ($suKey === false) {
+                array_unshift($scopes, 'su');
```

Comment:
This code is part of using Persona remotely to validate the token. The token is part of the url & a query parameter called `scope` is used to validate any of the listed scopes against the token. As `su` overrides all other scopes, it must be appended to the list in all cases.